src/routes/exercise-finished/ExerciseFinishedScreen.tsx

```diff
@@ -56,66 +67,103 @@ interface Props {
-    const correctResults = results.filter(doc => doc.result === 'correct')
-    const correct = correctResults.length / results.length
-    if (unlockedNextExercise) {
-      setMessage(labels.results.unlockedExercise)
-    } else if (correct > 2 / 3) {
-      setMessage(labels.results.feedbackGood)
-    } else if (correct > 1 / 3) {
-      setMessage(labels.results.feedbackMedium)
-    } else {
-      setMessage(labels.results.feedbackBad)
-    }
-  }, [results])
+  const [message, setMessage] = useState<string>('')
+  const [resultColor, setResultColor] = useState<Color>(theme.colors.background)
+  const [buttonText, setButtonText] = useState<string>('')
+  const [navigationAction, setNavigationAction] = useState<() => void>(() => undefined)
+  const [icon, setIcon] = useState<ReactElement>()
 
-  const repeatExercise = (): void => {
+  const repeatExercise = (): void =>
     navigation.navigate(EXERCISES[exercise].screen, {
       documents,
       disciplineId,
       disciplineTitle,
       closeExerciseAction
     })
+
+  const startNextExercise = (): void => {
+    if (exercise + 1 < EXERCISES.length) {
+      navigation.navigate(EXERCISES[exercise + 1].screen, {
+        documents,
+        disciplineId,
+        disciplineTitle,
+        closeExerciseAction
+      })
+    }
   }
 
-  const checkResults = (): void => {
-    navigation.navigate('Result', route.params)
+  const navigateToNextModule = (): void => {
+    navigation.pop(2)
   }
 
+  React.useEffect(() => {
+    const isLastExercise = exercise === EXERCISES.length - 1
+    const iconSize = wp('10%')
+    if (unlockedNextExercise && !isLastExercise) {
+      setMessage(
+        `${labels.results.unlockExercise.part1} ${EXERCISES[exercise + 1].title} ${labels.results.unlockExercise.part2}`
+      )
+      setResultColor(theme.colors.primary)
+      setButtonText(labels.results.button.nextExercise)
+      setNavigationAction(() => () => startNextExercise())
+      setIcon(<OpenLockIcon width={iconSize} height={iconSize} />)
+    } else if (percentageOfCorrectResults > 1 / 3) {
+      setResultColor(theme.colors.correct)
+      if (!isLastExercise) {
+        setMessage(labels.results.feedbackGood)
+        setButtonText(labels.results.button.continue)
+        setNavigationAction(() => () => startNextExercise())
+        setIcon(<HappSmileyIcon width={iconSize} height={iconSize} />)
+      } else {
+        setMessage(labels.results.finishedModule)
+        setButtonText(labels.results.button.close)
+        setNavigationAction(() => () => navigateToNextModule())
+        setIcon(<PartyHornIcon width={iconSize} height={iconSize} />)
+      }
+    } else {
+      setMessage(labels.results.feedbackBad)
+      setResultColor(theme.colors.incorrect)
+      setButtonText(labels.results.button.repeat)
+      setNavigationAction(() => () => repeatExercise())
+      setIcon(<SadSmileyIcon width={iconSize} height={iconSize} />)
+    }
+  }, [results])
+
   return (
     <Root>
```

Comment:
the `root` container has to be wrapped with a SafeAreaView to get the close button clickable for ios
```js
    <SafeAreaView style={{flex:1}}>
```
probably we have to wrap every screen with that container to ensure working all the ui elements on ios devices with notches. I can create a ticket for that. It's important not to wrap the navigation stuff just fyi

Reply:
Yes please open a ticket. Will fix this screen now, nevertheless.